src/silx/utils/number.py

```diff
@@ -105,13 +106,17 @@ def min_numerical_convertible_type(string, check_accuracy=True):
         exponent = "0"
 
     nb_precision_digits = int(exponent) - len(decimal) - 1
-    precision = _biggest_float(10) ** nb_precision_digits * 1.2
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", "overflow encountered in scalar power", RuntimeWarning)
+        precision = _biggest_float(10) ** nb_precision_digits * 1.2
     previous_type = _biggest_float
     for numpy_type in _float_types:
         if numpy_type == _biggest_float:
             # value was already casted using the bigger type
             continue
-        reduced_value = numpy_type(value)
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", "overflow encountered in cast", RuntimeWarning)
+            reduced_value = numpy_type(value)
```

Comment:
Having tests for raise those warnings, I expects that overflows were taken into accounts when writing the code 🤷

Reply:
The docstring says "It has some known issues: precision loss." so I guess.